benchmark/benchmark.cpp

```diff
@@ -62,73 +69,63 @@ struct proxy_pool : public pool_interface<Provider> {
     getOps([[maybe_unused]] ::benchmark::State &state) override {
         return umfProxyPoolOps();
     }
-    void *getParams([[maybe_unused]] ::benchmark::State &state) override {
-        return nullptr;
-    }
+
     static std::string name() { return "proxy_pool<" + Provider::name() + ">"; }
 };
 
 #ifdef UMF_POOL_DISJOINT_ENABLED
 template <typename Provider>
 struct disjoint_pool : public pool_interface<Provider> {
-    umf_disjoint_pool_params_handle_t disjoint_memory_pool_params;
+    umf_memory_pool_ops_t *
+    getOps([[maybe_unused]] ::benchmark::State &state) override {
+        return umfDisjointPoolOps();
+    }
 
-    disjoint_pool() {
-        disjoint_memory_pool_params = NULL;
-        auto ret = umfDisjointPoolParamsCreate(&disjoint_memory_pool_params);
+    typename pool_interface<Provider>::params_ptr
+    getParams(::benchmark::State &state) override {
+        umf_disjoint_pool_params_handle_t raw_params = nullptr;
+        auto ret = umfDisjointPoolParamsCreate(&raw_params);
         if (ret != UMF_RESULT_SUCCESS) {
-            return;
+            state.SkipWithError("Failed to create disjoint pool params");
+            return {nullptr, [](void *) {}};
         }
 
-        // those function should never fail, so error handling is minimal.
-        ret = umfDisjointPoolParamsSetSlabMinSize(disjoint_memory_pool_params,
-                                                  4096);
+        ret = umfDisjointPoolParamsSetSlabMinSize(raw_params, 4096);
         if (ret != UMF_RESULT_SUCCESS) {
-            goto err;
+            state.SkipWithError("Failed to set slab min size");
+            umfDisjointPoolParamsDestroy(raw_params);
+            return {nullptr, [](void *) {}};
         }
 
-        ret = umfDisjointPoolParamsSetCapacity(disjoint_memory_pool_params, 4);
+        ret = umfDisjointPoolParamsSetCapacity(raw_params, 4);
         if (ret != UMF_RESULT_SUCCESS) {
-            goto err;
+            state.SkipWithError("Failed to set capacity");
+            umfDisjointPoolParamsDestroy(raw_params);
```

Comment:
Instead of calling `umfDisjointPoolParamsDestroy` every time an error code is returned by some setter function, it is better to use a scoped object like `unique_ptr` that destroys the underlying object automatically when you exist from the scope.

Reply:
done.